### PR TITLE
fix(autodiscovery) : Fix TextField while debouncing value

### PR DESCRIPTION
**Pull request by Med1233**
## Description

we can't type text in providers' filter search field 

the problem is that we shouldn't use the value property with debounced TextField

preview : 

https://user-images.githubusercontent.com/109956462/230473441-f359ef8c-77db-417b-b7ef-864f2055e608.mp4

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x (master)

<h2> How this pull request can be tested ? </h2>

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Review by Thebarda — COMMENTED** [review]
I think it could be clever to handle the debounce within the TextField component.
You can set it via a prop name `debounce?: number`

**Comment by Med1233** [comment]
> I think it could be clever to handle the debounce within the TextField component. You can set it via a prop name `debounce?: number`

debounce function uses the setValue function which is not defined in the TextField but in the component that uses the TextField. And this prob happened just because we shouldn't use value Props or defaultValue with debounce

**Comment by Med1233** [comment]
> > I think it could be clever to handle the debounce within the TextField component. You can set it via a prop name `debounce?: number`
> 
> debounce function uses the setValue function which is not defined in the TextField but in the component that uses the TextField. And this prob happened just because we shouldn't use value Props or defaultValue with debounce

**Comment by Thebarda** [comment]
Yes but you can set it trough the setInnerValue. So that when you change the value, it calls the setInnerValue and the onChange prop's function. But we could talk about it later because I do think it's better to create a DebounceTextField that handle debounced value

File: centreon/packages/ui/src/InputField/Text/index.tsx [hunk 1]
```diff
@@ -123,7 +125,17 @@ const TextField = forwardRef(
...
+      if (debounced) {
+        return {};
+      }
+
+      if (defaultValue) {
+        return { defaultValue };
+      }
+
+      return { value: innerValue };
+    };
```
Comment (Thebarda) [inline centreon/packages/ui/src/InputField/Text/index.tsx:138]:
```suggestion
    const getValueProps = useCallback((): object => {
      if (debounced) {
        return {};
      }

      if (defaultValue) {
        return { defaultValue };
      }

      return { value: innerValue };
    }, [innerValue, debounced, defaultValue]);
```

**Review by Thebarda — APPROVED** [review]

**Pull request merged**